src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_copy_b.cpp

```diff
@@ -57,39 +57,31 @@ void BrgemmCopyB::custom_constructor_validate_and_infer_types(std::vector<size_t
     // During ctor call, BrgemmCopyB doesn't know his port descriptors.
     // So we use port descs from source inputs
     const auto element_type = get_input_element_type(0);
-    const auto pshape = snippets::utils::get_planar_pshape(get_input_partial_shape(0), layout_input);
-    validate(pshape, element_type);
+    const auto& pshape = get_input_partial_shape(0);
+    const auto planar_pshape = snippets::utils::get_planar_pshape(pshape, layout_input);
+    set_output_type(0, element_type, planar_pshape);
+    if (is_with_compensations()) {
```

Comment:
Minor: this is not a straightforward behavior, so let's add a comment here on why it's done this way

Reply:
Done, thanks